pkg/util/test/util.go

```diff
@@ -1043,6 +1069,76 @@ func ValidateAutopilot(pxImageList map[string]string, cluster *corev1.StorageClu
 	return nil
 }
 
+// ValidatePortworxProxy validates portworx proxy components
+func ValidatePortworxProxy(cluster *corev1.StorageCluster, timeout time.Duration) error {
+	proxyDs := &appsv1.DaemonSet{}
+	proxyDs.Name = "portworx-proxy"
+	proxyDs.Namespace = "kube-system"
+
+	pxService := &v1.Service{}
+	pxService.Name = "portworx-service"
+	pxService.Namespace = "kube-system"
+
+	if isPortworxProxyEnabled(cluster) {
+		logrus.Debug("Portworx proxy is enabled in StorageCluster")
+
+		// Validate Portworx proxy ServiceAccount
+		_, err := coreops.Instance().GetServiceAccount(proxyDs.Name, proxyDs.Namespace)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate ServiceAccount %s, Err: %v", proxyDs.Name, err)
+		}
+
+		// Validate Portworx proxy ClusterRoleBinding
+		_, err = rbacops.Instance().GetClusterRoleBinding(proxyDs.Name)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate ClusterRoleBinding %s, Err: %v", proxyDs.Name, err)
+		}
+
+		// Validate Portworx proxy Service in kube-system namespace
+		if err := validatePortworxService("kube-system"); err != nil {
+			return err
+		}
+		_, err = coreops.Instance().GetService(pxService.Name, pxService.Namespace)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate Service %s, Err: %v", pxService.Name, err)
+		}
+
+		// Validate Portworx proxy DaemonSet
+		if err := validateDaemonSet(proxyDs, timeout); err != nil {
+			return err
+		}
```

Comment:
Let's move this validation before the ServiceAccount validation. In case we call ValidatePortworxProxy directly, it might fail because the operator might not have started creating the proxy components. If we call validateDaemonSet before, it will wait for the daemonset to be ready. At that point the service account and other components would be created for sure.